kernel/common/enumerable.rb

```diff
@@ -351,7 +351,18 @@ def any?
   end
 
   def cycle(many=nil)
-    return to_enum(:cycle, many) unless block_given?
+    unless block_given?
+      size = Proc.new do
```

Comment:
This pattern occurs in a few places in this PR so this comment applies to all of them: can we not simply pass the size itself here instead of a Proc? There's little overhead in calculating the size at this point and it removes the need for evaluating the block on every `#size` call later on.